pkg/git/git.go

```diff
@@ -1,25 +1,104 @@
 package git
 
 import (
+	"encoding/json"
+	"fmt"
+	"net/url"
 	"os/exec"
+	"strings"
 )
 
+var Path = ""
+
+// HasUncommittedChanges checks repository for uncommitted changes
+func HasUncommittedChanges() bool {
+	if !IsGitRepository() {
+		return false
+	}
+
+	_, err := runGitCommand("diff", "--quiet", "HEAD")
+	return err != nil
+}
+
 // IsGitRepository checks if current directory is a git repository
 func IsGitRepository() bool {
-	_, err := exec.Command("git", "rev-parse", "--is-inside-working-tree").Output()
+	_, err := runGitCommand("rev-parse", "--is-inside-working-tree")
 	return err == nil
 }
 
-// HasUncommittedChanges checks repository for uncommitted changes
-func HasUncommittedChanges() bool {
-	if IsGitRepository() {
-		// There is no system independent way to get exit code
-		// so we treat all exit codes the same, this works as long as we check IsGitRepository first
-		_, err := exec.Command("git", "diff", "--quiet", "HEAD").Output()
-		if err != nil {
-			return true
-		}
+func GetRemoteRepository(remote string) (host, account, repo string, err error) {
```

Comment:
I feel we should probably have some unit tests of this parsing function please.

I think it will fail on ssh URLs like this one `﻿git@github.com:astronomer/astro-cli`

Reply:
TIL about Git having these "SCP-like" URLs.